packages/client/hmi-client/src/services/workflow.ts

```diff
@@ -38,6 +38,7 @@ export const addNode = (
 	wf: Workflow,
 	op: Operation,
 	pos: Position,
+	state: any = {},
 	size: Size = { width: 180, height: 220 }
```

Comment:
You can't just add a new parameter in the middle of a functional signature, it breaks other instances. e.g.

```
workflowService.addNode(wf.value, SimulateJuliaOperation, newNodePosition, {
	width: 420,
	height: 220
});

```

If you have more than one optional parameters, I would suggest you encapsulate them inside some kind of option object. Otherwise it becomes confusing as you have a situation where you need 4 arguments of different types.

Reply:
Just forgot to check other uses of `addNode`